growth/too/tasks/slack.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (C) 2015  Leo Singer
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+"""
+Some tools for parsing and composing slack messages.
+"""
+__author__ = "Michael Coughlin <michael.coughlin@ligo.org>"
+
+from flask import render_template
+from slack import WebClient
+
+from ..flask import app
+from . import celery
+from .. import models
+
+client = WebClient(token=app.config['SLACK_API_TOKEN'])
```

Comment:
Importing this module will fail unless this API token is defined. It won't be defined in development configurations. How would you like to deal with this?

Reply:
Is this fundamentally different than Twilio? maybe we just return? or something. I could use suggestions.

Reply:
It is different, because the Twilio credentials are set up lazily by [flask-twilio](https://flask-twilio.readthedocs.io/en/latest/) using the Flask application factory pattern (see https://flask.palletsprojects.com/en/1.1.x/patterns/appfactories/#factories-extensions for further explanation).

Reply:
So we could return an empty list for the WebClient if there is no token, and then just capture this when we call below?

Reply:
You could create a lazy singleton, like this:

```python
from celery.local import PromiseProxy
from slack import WebClient

from ..flask import app


def get_client():
    return WebClient(token=app.config['SLACK_API_TOKEN'])


client = PromiseProxy(get_client)
```